contracts/clients/CkbProof.sol

```diff
@@ -0,0 +1,342 @@
+    CommitmentKV[] commitments;
+    bytes content;
+}
+
+library CkbLightClient {
+    event GetHeaderEvent(CKBHeader);
+    event NotGetHeaderEvent();
+
+    function getHeader(bytes32 blockHash) public returns (CKBHeader memory) {
+        // axon_precompile_address(0x02)
+        address get_header_addr = address(0x0102);
+        (bool isSuccess, bytes memory res) = get_header_addr.staticcall(
+            abi.encode(blockHash)
+        );
+
+        CKBHeader memory header;
+        if (isSuccess) {
+            header = abi.decode(res, (CKBHeader));
+            /*
+            replace above decode into the following data can pass test
+            header = CKBHeader({
+                version: 0,
+                compactTarget: 0,
+                timestamp: 0,
+                number: 0,
+                epoch: 0,
+                parentHash: bytes32(0),
+                transactionsRoot: 0x7c57536c95df426f5477c344f8f949e4dfd25443d6f586b4f350ae3e4b870433,
+                proposalsHash: bytes32(0),
+                extraHash: bytes32(0),
+                dao: bytes32(0),
+                nonce: uint128(0),
+                extension: "",
+                blockHash: bytes32(0)
+            });
+            */
+            emit GetHeaderEvent(header);
+        } else {
+            emit NotGetHeaderEvent();
+        }
+        return header;
+    }
+}
+
+using CkbLightClient for bytes32;
+
+// Define ckb blake2b
+function blake2b(bytes memory data) view returns (bytes32) {
+    // axon_precompile_address(0x06)
+    address blake2b_addr = address(0x0106);
+    (bool isSuccess, bytes memory res) = blake2b_addr.staticcall(data);
+
+    bytes32 hash;
+    if (isSuccess) {
+        hash = abi.decode(res, (bytes32));
+    }
+    return hash;
+}
+
+function ckbMbtVerify(VerifyProofPayload memory payload) view returns (bool) {
+    // axon_precompile_address(0x07)
+    address ckb_mbt_addr = address(0x0107);
+    (, bytes memory res) = ckb_mbt_addr.staticcall(
+        abi.encode(
+            payload.verifyType,
+            payload.transactionsRoot,
+            payload.witnessesRoot,
+            payload.rawTransactionsRoot,
+            payload.proof
+        )
+    );
+
+    return uint8(res[0]) == 1;
+}
+
+function calculateHashes(
+    bytes memory ckbTransaction
+) view returns (bytes32 transactionHash, bytes32 witnessHash) {
+    // Calculate the hashes here
+    (uint256 offset, uint256 size) = ckbTransaction.readCKBTxRaw();
+    bytes memory raw_tx = new bytes(size);
+    for (uint i = 0; i < size; i++) {
+        raw_tx[i] = ckbTransaction[i + offset];
+    }
+
+    return (blake2b(raw_tx), blake2b(ckbTransaction));
+}
+
+function decodeRlpEnvelope(
+    bytes memory rlpEncodedData
+) pure returns (Envelope memory) {
+    RLPReader.RLPItem[] memory ls = rlpEncodedData.toRlpItem().toList();
+
+    // Decode the msg_type
+    // MsgType msg_type = MsgType(ls[0].toUint());
+    MsgType msg_type = MsgType.MsgClientCreate;
```

Comment:
IMO having a wrong `msg_type` is worse than ignoring it. I suggest that we just remove `msg_type` in `Envelope` for now.